daemon/accesslog.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io/ioutil"
+
+	"github.com/cilium/cilium/api/v1/models"
+	restapi "github.com/cilium/cilium/api/v1/server/restapi/daemon"
+	"github.com/cilium/cilium/pkg/apierror"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/spf13/viper"
+)
+
+type getAccesslog struct {
+	daemon *Daemon
+}
+
+const accesLogKey = "access-log"
+
+// NewGetAccesslogHandler returns the accesslog endpoint handler for the agent
+func NewGetAccesslogHandler(d *Daemon) restapi.GetAccesslogHandler {
+	return &getAccesslog{daemon: d}
+}
+
+func (h *getAccesslog) Handle(params restapi.GetAccesslogParams) middleware.Responder {
+	dr := models.Accesslog{}
+
+	path := viper.GetString(accesLogKey)
+	dr.Path = path
+
+	content, err := ioutil.ReadFile(path)
```

Comment:
I don't think that this is a good idea.
1) I'm not sure it makes sense to expose the full access log via the API at this point.
2) If we want to add this, it should be added as a streaming API where only the access log stream is returned.

I would return the path via `DaemonConfigurationResponse` as part of the `GET /config` API

Reply:
Dropping this then. I can write follow up issues later.